contracts/v0.6/SimpleSwap.sol

```diff
@@ -0,0 +1,178 @@
+pragma solidity ^0.6.0;
+
+import "@chainlink/contracts/src/v0.6/Owned.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./token/ERC677Receiver.sol";
+
+contract SimpleSwap is Owned, ReentrancyGuard {
+  using SafeMath for uint256;
+
+  event LiquidityUpdated(
+    uint256 indexed amount,
+    address indexed source,
+    address indexed target
+  );
+  event TokensSwapped(
+    uint256 indexed amount,
+    address indexed source,
+    address indexed target,
+    address caller
+  );
+
+  mapping(address => mapping(address => uint256)) private s_swappableAmount;
+
+  /**
+   * @dev deposits tokens from the target of a swap pair but does not return any
+   * @param amount count of liquidity being added
+   * @param source the token that can be swapped for what is being deposited
+   * @param target the token that can is being deposited for swapping
+   */
+  function addLiquidity(
```

Comment:
One potential concern with this function being world-callable is that somebody could add liquidity coming from a malicious token contract, making this contract interact with a token that doesn't conform to ERC20.
Based on my first pass over the code, I believe that we handle this case correctly.
However, I would be more comfortable with a more conservative approach of **only interacting with tokens that have been allowlisted** by the `owner`. This would also prevent a scammer from convincing users to interact with a malicious token through this "Chainlink-endorsed-and-deployed" contract.

## Suggested allowlisting approach:
Add a `mapping(address => bool) s_tokenAllowlist` plus a public getter and an onlyOwner setter.

In `_addLiquidity`, add some `require`s:
```
    require(s_tokenAllowlist[source], "source not allowlisted");
    require(s_tokenAllowlist[target], "target not allowlisted");

    uint256 newAmount = getSwappableAmount(source, target).add(amount);
    s_swappableAmount[source][target] = newAmount;

    emit LiquidityUpdated(newAmount, source, target);
```